### ISSUE TITLE
Request for a Polygon snapshot update

### ISSUE THREAD

**Issue by blockchen-admin** [maticnetwork/bor#1251]
The mainnet snapshot data uploaded on https://snapshots.polygon.technology/ has not been updated since February 1st. It takes a considerable amount of time to download and then update to the latest data. Could you please update it with the latest data?

**Comment by Zorato** [comment]
Would be nice to have snapshot with state scheme path & pebble db as db engine.

**Comment by sammy007** [comment]
Basically polygon team lied to everybody in the start of 2024 https://polygon.technology/blog/empowering-the-blockchain-community-automated-snapshot-optimization-and-ecosystem-advancements

It requires little resources, almost peanuts for a well funded project. Refusing to provide "official" snapshots is an infrastructural irresponsibility. None of the validators (aka community) neither other parties are really interested in providing "community snapshots", it's basically a conflict of the interests.

**Comment by 0xsajal** [comment]
Hey,

A forum post was released earlier where we announced that snapshots will be provided by the community moving forward: https://forum.polygon.technology/t/stakecraft-introduces-a-new-service-for-polygon-community-all4nodes-io-aggregator-service/13694

You can go [here](https://all4nodes.io/Polygon) and find all the snaphsots uploaded by the community, including the snapshots last uploaded by Polygon.
There are multiple options available to download here. Also, please ignore the exclamation under status and click on the source to find a step-by-step guide to download the snapshots.

Hope this helps!

Thank You.

**Comment by sammy007** [comment]
Writing off snapshots on community is "passing the buck" on parties with conflict of interest. Validators are competitors to each other, nobody is interested in providing services. You will have few friendly nerds uploading them, the service will be unreliable and with thin bandwidth, just like this. Neither it's possible to verify the integrity of snapshots. It will take forever to obtain PBSS from so called community. Polygon should consider resuming service and uploading PBSS snapshots.

**Comment by 0xsajal** [comment]
By community I was referring to the validators here. All the snapshots you see in the link I shared above are provided by different validators. If you have problems with using any snapshot you can let us know, and we will make sure that is addressed. 
Also, PBSS is currently under testing and when it releases on PoS, we will work with all the people responsible for generating and maintaining snapshots to ensure they are available at the earliest.

**Comment by valamidev** [comment]
> By community I was referring to the validators here. All the snapshots you see in the link I shared above are provided by different validators. If you have problems with using any snapshot you can let us know, and we will make sure that is addressed. Also, PBSS is currently under testing and when it releases on PoS, we will work with all the people responsible for generating and maintaining snapshots to ensure they are available at the earliest.

What about the developers? :'( Not very much appealing, that running a Bor cost at least 2x more than any other existing EVM chain (it is closer to 3x, than 2x).  

I know that PBSS is not a silver bullet which give a 10x improve on this side, but I also do not feel that it has any real priority.

**Comment by sammy007** [comment]
Moreover, PBSS requires sync from scratch so "as part of testing" the team can do it and upload snapshot.

**Comment by sammy007** [comment]
18h of downtime to compact chaindata from 7TB size to 5.0T and still in progress.
```
Iterated snapshot                        accounts=330,153,190 slots=5,010,507,483 elapsed=2h53m3.966s
Pruning state data                       nodes=5,010,523,868 skipped=10,010,650,984 size=1.63TiB    elapsed=4h42m41.090s eta=8.582s
Compacting database                      range=0x80-0x90 elapsed=10h8m4.720s
```

This is unbearable routing and PBSS snapshot is urgently needed to finally get rid of that procedure.

```
Xeon E-2338G, 128GB RAM.
SSD Intel P4510 8ТБ U.2 SSDPE2KX080T801 PCIe 3D NAND TLC
```

**Comment by sammy007** [comment]
Interesting is that all4nodes snapshots are of size 3.5 TB. After pruning my own chaindata I got 4.5 TB. Something is wrong obviously.

**Comment by ssandeep** [comment]
Although Bor is a fork of Geth, there are some changes which are unique to Bor and hence it warrants thorough testing, especially for big changes like introducing a new storage scheme. The process to test PBSS has already been started, which involves syncing the chain from genesis. As @0xsajal mentioned in the above post, once this node is fully synced on Mainnet, running various test cases and after careful monitoring for a few weeks after syncing, PBSS snapshots will be made available. Of course developers/enthusiasts who are confident of running upstream code untested can go ahead and use the feature on their nodes even earlier.

**Comment by blockchen-admin** [comment]
@ssandeep 
When will the snapshot data converted to PBSS be updated?

**Comment by 0x090909** [comment]
@sammy007 you can use the block pruner as well to save 70% of space.

**Comment by sammy007** [comment]
> @sammy007 you can use the block pruner as well to save 70% of space.

I do not want to use block pruner because I need full node, not a castrated node

**Comment by sammy007** [comment]
So as predicted, we have castrated pruned snapshot from what you call a community, the rest of snapshots are simply not available.

**Comment by VSGic** [comment]
So, I did not understand, community snapshots are pruned?

**Comment by ssandeep** [comment]
@blockchen-admin, the node is at ~50M blocks and still catching up. So it will take another 12-15 days to catch up to the tip of the chain at the current rate. But as I mentioned earlier, once the node is fully synced, we want to let it run for a few more weeks and run some tests before taking the snapshot.

**Comment by kaber2** [comment]
> So as predicted, we have castrated pruned snapshot from what you call a community, the rest of snapshots are simply not available.

Polygon has obviously given up on this blockchain. After lamenting for over a year over data availability wrt. chaindata pruning, they don't even offer snapshots themselves anymore, while syncing from scratch is impossible. A complete joke.

Similar with PBSS, they claim it needs to tested, yet users can not test it due to a lack of snapshots. This demonstrates a very non open mindset. Sure, they should do testing too, but the majority of bugs is usually found in the field, by actual users.

So in conclusion, they seemed to have lined their pockets enough and don't even want to spend the little amount that is required to provide their operators with the data they need, after letting the blockchain bloat like crazy for years. This should be a lesson for everyone not to trust any further developments coming from Polygon Technology, as they either don't understand how to properly run a community project or they just don't care.

**Comment by 0x090909** [comment]
@kaber2 I do have a pruned snapshot, around 1.1 TB total size

**Comment by kaber2** [comment]
> @kaber2 I do have a pruned snapshot, around 1.1 TB total size

Thanks, but I have given up on Polygon and deactivated my nodes a while ago. But just as multiple of the commenters here, I used to require full snapshots for at least some of my nodes. After hearing for a year about data availability, these seem to be unavailable nowadays thanks to Polygon not providing them anymore.

**Comment by longlaogithub** [comment]
> Hey,
> 
> A forum post was released earlier where we announced that snapshots will be provided by the community moving forward: https://forum.polygon.technology/t/stakecraft-introduces-a-new-service-for-polygon-community-all4nodes-io-aggregator-service/13694
> 
> You can go [here](https://all4nodes.io/Polygon) and find all the snaphsots uploaded by the community, including the snapshots last uploaded by Polygon. There are multiple options available to download here. Also, please ignore the exclamation under status and click on the source to find a step-by-step guide to download the snapshots.
> 
> Hope this helps!
> 
> Thank You.

That help!

**Comment by Zorato** [comment]
> @blockchen-admin, the node is at ~50M blocks and still catching up. So it will take another 12-15 days to catch up to the tip of the chain at the current rate. But as I mentioned earlier, once the node is fully synced, we want to let it run for a few more weeks and run some tests before taking the snapshot.

Any updates on this one?

**Comment by anshalshukla** [comment]
hey guys, we have a pebble+pbss snapshot available now and can be accessed [here](https://publicnode.com/snapshots#polygon). This is still in beta phase and we are still testing few things and the next version should fix some of the known issues.

**Comment by ashwani421** [comment]
so how to use this snapshot https://publicnode.com/snapshots#polygon
